pkg/destinations/redshift/insert.go

```diff
@@ -0,0 +1,170 @@
+package redshift
+
+import (
+	"path/filepath"
+
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+	"github.com/scratchdata/scratchdata/util"
+
+	"github.com/scratchdata/scratchdata/pkg/storage/blobstore/s3"
+)
+
+func (s *RedshiftServer) getGolumnNames(table string) (map[string]bool, error) {
+	schema := "public"
+	if s.Schema != "" {
+		schema = s.Schema
+	}
+
+	sql := `
+		select "column" as column_name
+		from pg_table_def
+		where schemaname = $1 and tablename = $2
+	`
+
+	m := map[string]bool{}
+	rows, err := s.conn.Query(sql, schema, table)
+	if err != nil {
+		return nil, fmt.Errorf("getGolumnNames: cannot fetch column names: %w", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+
+		name := ""
+		if err := rows.Scan(&name); err != nil {
+			return nil, fmt.Errorf("getGolumnNames: cannot scan column name: %w", err)
+		}
+		m[strings.ToLower(name)] = true
+	}
+	if len(m) == 0 {
+		return nil, fmt.Errorf("getGolumnNames: no columns found: %w", err)
+	}
+	return m, nil
+}
+
+func (s *RedshiftServer) createColumns(table string, jsonTypes map[string]string) error {
+	cols, err := s.getGolumnNames(table)
```

Comment:
No need for this. Just use IF NOT EXISTS in SQL when creating columns.

Reply:
@poundifdef not supported in aws redshift

Reply:
its weird though, it is supported while creating tables, but not altering one.